wrangler-service/src/main/java/co/cask/wrangler/service/hiveserver2/Hiveserver2Service.java

```diff
@@ -0,0 +1,482 @@
+       values.add(value);
+
+       JsonObject response = new JsonObject();
+       response.addProperty("status", HttpURLConnection.HTTP_OK);
+       response.addProperty("message", "Success");
+       response.addProperty("count", values.size());
+       response.add("values", values);
+       sendJson(responder, HttpURLConnection.HTTP_OK, response.toString());
+     } catch (Exception e) {
+       LOG.error("Error occurred while closing connection ", e);
+       error(responder, e.getMessage());
+     }
+   }
+
+   private JsonObject getHiveTableSchema(HttpServiceRequest request, String connectionURL, String dbName, String tableName){
+     JsonObject schema = null;
+
+     java.sql.Connection hiveConnection=null;
+     PreparedStatement ps = null;
+     ResultSet rs = null;
+     try {
+       String updatedConnectionURL = addDelegationTokenUpdateURL(connectionURL, request);
+       hiveConnection = DriverManager.getConnection(updatedConnectionURL);
+
+       ps = hiveConnection.prepareStatement(String.format("describe %s.%s", dbName, tableName));
+       rs = ps.executeQuery();
+
+       schema = new JsonObject();
+       schema.addProperty("type", "record");
+       schema.addProperty("name", "etlSchemaBody");
+
+       JsonArray fields = new JsonArray();
+       while (rs.next()) {
+         JsonObject column = new JsonObject();
+         column.addProperty("name", rs.getString("col_name"));
+         column.addProperty("type", rs.getString("data_type"));
+         fields.add(column);
+       }
+       schema.add("fields", fields);
+
+     } catch (Exception e) {
+       LOG.error("Error fetching hive table schema ", e);
+       e.printStackTrace();
+     }finally{
+       close(rs);
+       close(ps);
+       close(hiveConnection);
+     }
+
+     return schema;
+   }
+
+   private String addDelegationTokenUpdateURL(final String jdbcURL, HttpServiceRequest request) throws ClassNotFoundException, LoginException, IOException {
+     Class.forName("org.apache.hive.jdbc.HiveDriver");
+
+     StringBuilder updatedJdbcURL = new StringBuilder(jdbcURL);
+
+     UserGroupInformation ugi = Utils.getUGI();
+     if(ugi.isSecurityEnabled()){
+       if(updatedJdbcURL.indexOf("auth=")==-1) updatedJdbcURL.append(";auth=delegationToken");
+
+       Utils.setTokenStr(ugi,getHiveServer2ClientToken(), HiveAuthFactory.HS2_CLIENT_TOKEN);
+
+       // User impersonation
+       if (getContext().getRuntimeArguments().containsKey("system.user.impersonation.enabled")
+               && getContext().getRuntimeArguments().get("system.user.impersonation.enabled").equalsIgnoreCase("true")){
+         updatedJdbcURL.append(";hive.server2.proxy.user=").append(request.getHeader("CDAP-UserId"));
+       }
+     }
+
+     if(updatedJdbcURL.indexOf("?")==-1){
+       updatedJdbcURL.append("?hive.resultset.use.unique.column.names=false");
+     }else{
+       updatedJdbcURL.append("&hive.resultset.use.unique.column.names=false");
```

Comment:
Please check if we can retrieve Hive property names from some Hive java class. Its better to use constants rather than harcoding in code. If there's no such Property name then please create a constant at the top of your class as final static.

Reply:
getting below error: 
Caused by: java.lang.ClassNotFoundException: org.apache.hadoop.mapreduce.TaskAttemptContext

keeping it as it is.